### PR TITLE
Adding CountDownLatch config

### PR DESCRIPTION
**Pull request by tombujok**
Also fixing omissions in AtomicReference and AtomicLong configs.

Compliant with #11913 

Client side will be dealt with in another PR (tracked here: #11876)

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java [hunk 1]
```diff
@@ -133,6 +136,9 @@
             lockConfigs,
             listConfigs,
             setConfigs,
+            atomicLongConfigs,
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java]: Why didn't we see any failures with the missing `atomicLongConfigs` and `atomicReferenceConfigs` before?

The field seems to be used in the context of split-brain healing, which collects all existing dynamic configs and resets the service. Maybe we need a `DynamicConfigSplitBrainTest` which checks that all kind of configurations are correctly merged.

File: hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java [hunk 2]
```diff
@@ -271,6 +277,15 @@ public void registerConfigLocally(IdentifiedDataSerializable newConfig,
         } else if (newConfig instanceof LockConfig) {
             LockConfig lockConfig = (LockConfig) newConfig;
             currentConfig = lockConfigs.putIfAbsent(lockConfig.getName(), lockConfig);
+        } else if (newConfig instanceof AtomicLongConfig) {
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java]: Why didn't we see any failures for the missing `AtomicLongConfig` and `AtomicReferenceConfig` before?  Are we missing a test?

File: hazelcast/src/main/resources/hazelcast-config-3.10.xsd [hunk 3]
```diff
@@ -1323,6 +1324,21 @@
             </xs:simpleType>
         </xs:attribute>
     </xs:complexType>
+    <xs:complexType name="count-down-latch">
+        <xs:all>
+            <xs:element name="quorum-ref" minOccurs="0" maxOccurs="1"/>
```
Comment (Donnerbart) [inline hazelcast/src/main/resources/hazelcast-config-3.10.xsd]: Shouldn't we specify a type like `type="xs:string"` here? We didn't do this for any `quorum-ref` so far, but I'm not sure what the default type is.

File: hazelcast/src/test/java/com/hazelcast/config/ConfigCompatibilityChecker.java [hunk 4]
```diff
@@ -325,6 +326,21 @@ SemaphoreConfig getDefault(Config c) {
...
+                    && nullSafeEqual(c1.getQuorumName(), c2.getQuorumName());
+        }
+
+        @Override
+        CountDownLatchConfig getDefault(Config c) {
+            return c.getCountDownLatchConfig("default");
+        }
+    }
+
+
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/config/ConfigCompatibilityChecker.java]: Superfluous newline.

File: hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java (hunk 1 above)
Comment (tombujok) [reply hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java]: That's a good question to @vbekiaris but it's out of scope of this PR, so plz address any changes related to it in another PR.

File: hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java (hunk 2 above)
Comment (tombujok) [reply hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java]: That's a good question to @vbekiaris but it's out of scope of this PR, so plz address any changes related to it in another PR.

File: hazelcast/src/main/resources/hazelcast-config-3.10.xsd (hunk 3 above)
Comment (tombujok) [reply hazelcast/src/main/resources/hazelcast-config-3.10.xsd]: I just followed the other quorum-ref definitions. Don't want to change it to a different type in one part and have it done differently in the older parts. Could be cleaned up in 4.0 IMHO.

File: hazelcast/src/test/java/com/hazelcast/config/ConfigCompatibilityChecker.java (hunk 4 above)
Comment (tombujok) [reply hazelcast/src/test/java/com/hazelcast/config/ConfigCompatibilityChecker.java]: thx, fixed.

**Comment by tombujok** [comment]
run-lab-run

**Comment by tombujok** [comment]
run-lab-run

**Comment by tombujok** [comment]
run-lab-run

**Pull request merged**